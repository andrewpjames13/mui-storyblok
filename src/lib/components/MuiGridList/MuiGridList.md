MuiGridList Example:

```js
const yoda = 'https://d.newsweek.com/en/full/1571628/baby-yoda.webp?w=790&f=163be219fae1ece8c6cbb38755c119b2';
const cupcake = 'https://www.lifeloveandsugar.com/wp-content/uploads/2018/01/Chocolate-Covered-Strawberry-Cupcakes4.jpg';
const content = [
  {
    cols: '2',
    component: 'MuiTileData',
    image: cupcake,
    rows: '1',
    title: 'Cupcake',
    MuiGridListTileBar: [],
  },
  {
    cols: '1',
    component: 'MuiTileData',
    image: yoda,
    rows: '1',
    title: 'Yoda',
    MuiGridListTileBar: [],
  },
  {
    cols: '1',
    component: 'MuiTileData',
    image: cupcake,
    rows: '1',
    title: 'Cupcake',
    MuiGridListTileBar: [],
  },
  {
    cols: '2',
    component: 'MuiTileData',
    image: yoda,
    rows: '1',
    title: 'Yoda',
    MuiGridListTileBar: [],
  },
];


<MuiGridList
  cellHeight={180}
  cols={2}
  spacing={4}
  height={550}
  width={450}
  content={content}
/>
```

MuiGridList with MuiGridListTileBar:

```js
const yoda = 'https://d.newsweek.com/en/full/1571628/baby-yoda.webp?w=790&f=163be219fae1ece8c6cbb38755c119b2';
const cupcake = 'https://www.lifeloveandsugar.com/wp-content/uploads/2018/01/Chocolate-Covered-Strawberry-Cupcakes4.jpg';
const content = [
  {
    cols: '2',
    component: 'MuiTileData',
    image: cupcake,
    rows: '1',
    title: 'Cupcake',
    MuiGridListTileBar: [
      {
        component: 'MuiGridListTileBar',
        subtitle: '2 Col Span',
        title: 'A Grid Item',
        titlePosition: 'bottom',
        actionIcon: [
          {
            component: 'MuiIconButtonRedirect',
            redirectRoute: '',
            iconButton: [
              {
                component: 'MuiIconButton',
                icon: [
                  {
                    component: 'MuiIcon',
                    iconName: 'home',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
  {
    cols: '1',
    component: 'MuiTileData',
    image: yoda,
    rows: '1',
    title: 'Yoda',
    MuiGridListTileBar: [
      {
        component: 'MuiGridListTileBar',
        subtitle: '1 Col Span',
        title: 'Yoda Item',
        titlePosition: 'bottom',
        actionIcon: [
          {
            component: 'MuiIconButtonRedirect',
            redirectRoute: '',
            iconButton: [
              {
                component: 'MuiIconButton',
                icon: [
                  {
                    component: 'MuiIcon',
                    iconName: 'menu',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
  {
    cols: '1',
    component: 'MuiTileData',
    image: cupcake,
    rows: '1',
    title: 'Cupcake',
    MuiGridListTileBar: [
      {
        component: 'MuiGridListTileBar',
        subtitle: 'A Subtitle',
        title: 'Cupcake Grid',
        titlePosition: 'bottom',
        actionIcon: [
          {
            component: 'MuiIconButtonRedirect',
            redirectRoute: '',
            iconButton: [
              {
                component: 'MuiIconButton',
                icon: [
                  {
                    component: 'MuiIcon',
                    iconName: 'close',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
  {
    cols: '2',
    component: 'MuiTileData',
    image: yoda,
    rows: '1',
    title: 'Yoda',
    MuiGridListTileBar: [
      {
        component: 'MuiGridListTileBar',
        subtitle: 'A Subtitle',
        title: 'Yoda Grid',
        titlePosition: 'bottom',
        actionIcon: [
          {
            component: 'MuiIconButtonRedirect',
            redirectRoute: '',
            iconButton: [
              {
                component: 'MuiIconButton',
                icon: [
                  {
                    component: 'MuiIcon',
                    iconName: 'menu',
                  },
                ],
              },
            ],
          },
        ],
      },
    ],
  },
];


<MuiGridList
  cellHeight={180}
  cols={2}
  spacing={4}
  height={550}
  width={450}
  content={content}
/>
```