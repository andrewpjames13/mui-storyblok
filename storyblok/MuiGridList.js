module.exports = {
  "name": "MuiGridList",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:01.085Z",
  "schema": {
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiTileData"
      ],
      "required": false
    },
    "cellHeight": {
      "type": "number",
      "description": "* mui prop: number | 'auto'\n* Number of px for one cell height.\n* You can set 'auto' if you want to let the children determine the height.",
      "display_name": "",
      "default_value": "180"
    },
    "cols": {
      "type": "number",
      "description": "* mui prop: number\n* Number of columns.",
      "default_value": "2"
    },
    "spacing": {
      "type": "number",
      "description": "* mui prop: number\n* Number of px for the spacing between tiles.",
      "default_value": "4"
    },
    "rootClass": {
      "type": "options",
      "options": [],
      "description": "* storyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component. See CSS API below for more details.",
      "min_options": ""
    },
    "height": {
      "type": "text",
      "description": "Height of the gridList container",
      "max_length": "",
      "default_value": "500"
    },
    "width": {
      "type": "text",
      "description": "Width of the GridList container.",
      "default_value": "450"
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiGridList",
  "component_group_uuid": null
}