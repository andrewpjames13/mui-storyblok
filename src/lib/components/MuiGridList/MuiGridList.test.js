import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import MuiGridList from './MuiGridList';

function setup() {
  const props = {
    cellHeight: '180',
    cols: '3',
    spacing: '4',
    width: '400',
    height: '500',
    content: [
      {
        MuiGridListTileBar: [
          {
            actionIcon: [
              {
                component: 'MuiIconButtonRedirect',
                iconButton: [
                  {
                    component: 'MuiIconButton',
                    icon: [
                      {
                        component: 'MuiIcon',
                        iconName: 'home',
                      },
                    ],
                  },
                ],
                redirectRoute: 'home',
              },
            ],
            component: 'MuiGridListTileBar',
            subtitle: 'test1',
            title: 'home',
            titlePosition: 'bottom',
          },
        ],
        cols: '1',
        component: 'MuiTileData',
        image: 'www.test.com',
        rows: '1',
        title: 'bubble gum',
      },
    ],
  };

  const comp = shallow(<MuiGridList {...props} />);
  return { comp, props };
}

describe('<MuiGridList />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(
      <MemoryRouter>
        <MuiGridList {...props} />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiGridList', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
