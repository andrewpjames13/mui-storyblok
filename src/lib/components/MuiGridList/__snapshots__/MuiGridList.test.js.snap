// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiGridList /> snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <ul
    className="MuiGridList-root makeStyles-root-4"
    style={
      Object {
        "margin": -2,
      }
    }
  >
    <li
      className="MuiGridListTile-root"
      style={
        Object {
          "height": 184,
          "padding": 2,
          "width": "33.333333333333336%",
        }
      }
    >
      <div
        className="MuiGridListTile-tile"
      >
        <img
          alt="bubble gum"
          src="www.test.com"
        />
        <div
          className="MuiGridListTileBar-root makeStyles-root-10 MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
        >
          <div
            className="MuiGridListTileBar-titleWrap MuiGridListTileBar-titleWrapActionPosRight"
          >
            <div
              className="MuiGridListTileBar-title"
            >
              home
            </div>
            <div
              className="MuiGridListTileBar-subtitle"
            >
              <span>
                test1
              </span>
            </div>
          </div>
          <div
            className="MuiGridListTileBar-actionIcon"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root makeStyles-root-22"
              data-testid="muiIconButton"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <span
                  aria-hidden={true}
                  className="material-icons MuiIcon-root makeStyles-root-35 MuiIcon-colorSecondary"
                >
                  home
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
